File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        two_sum_pair  = {}

        for i, num in enumerate(nums):
            if target - num in two_sum_pair:
                return [i, two_sum_pair[target - num]]
            two_sum_pair[num] = i    

File: 0001-two-sum/0001-two-sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
     
     Map<int, int> seen = {};

     for(int i = 0 ; i < nums.length ; i++) {
         int comp = target - nums[i];
         if(seen.containsKey(comp)) {
            return [seen[comp]!, i];
         }
        seen[nums[i]] = i;
     }
    return []; 
  }
}